preprocessors/openstreetmap/osm_service.py

```diff
@@ -514,6 +430,39 @@ def compute_new_node(node_params, bearing, bbox_coordinates):
     return new_node
 
 
+# Get the latitude and the longitude of the new (immediate) node
+
+
+def get_new_node_coordinates(flag, lat1, lon1, var2, theta):
+    if flag:  # if true, solve for longitude
+        # Latitude in radians
+        lat2 = var2
+        # Longitude in radians
+        lon2 = ((math.tan(theta) * (lat2 - lat1)) / math.cos(lat1)) + lon1
+        # Convert lat/lon back to degrees
+        lat2 = degrees(lat2)
+        lon2 = degrees(lon2)
+    else:  # solve for latitude
+        # Longitude in radians
+        lon2 = var2
+        # Latitude in radians
+        lat2 = ((lon2 - lon1) * math.cos(lat1) / math.tan(theta))
+        # Convert latitude/longitude back to degrees
+        lat2 = degrees(lat2 + lat1)
+        lon2 = degrees(lon2)
+    return lat2, lon2
+
+
+def validate_new_node_coordinates(lat2, lon2, bbox_coordinates):
```

Comment:
If this validation is important, why is it only run in some cases? Why not check after generating _all_ new points?